e2e/rdctl.e2e.spec.ts

```diff
@@ -1165,13 +1169,11 @@ test.describe('Command server', () => {
     test('should verify nerdctl can talk to containerd', async() => {
       const { stdout } = await rdctl(['list-settings']);
       const settings: Settings = JSON.parse(stdout);
-      const payloadObject: RecursivePartial<Settings> = { };
 
-      if (settings.kubernetes.containerEngine !== ContainerEngine.CONTAINERD) {
-        payloadObject.kubernetes = {};
-        payloadObject.kubernetes.containerEngine = ContainerEngine.CONTAINERD;
-      }
-      if (Object.keys(payloadObject).length > 0) {
+      if (settings.containerEngine.name !== ContainerEngine.CONTAINERD) {
+        const payloadObject = {
```

Comment:
I'd like to see `const payloadObject: RecursivePartial<Settings> = {` just to ensure we have the type correct here (e.g. for when we change settings structure again). But I definitely like the new code flow better.